Adding tests for deprecation message when a REST wrapper implementing plugin presents (#66827)

BigPandaToo opened the pull request:
In 7.13 we added a deprecation message when a security REST 
wrapper implementing plugin presents.
This change adding tests to:

-Test the right deprecation message logged when custom 
security plugin installs rest handler
-Test no deprecation message logged when elastic Security 
plugin installs rest handler

BigPandaToo committed:
Test

BigPandaToo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch '7.x' into Plugin_name_7_test

BigPandaToo committed:
Adding tests to #73249 change #73329

-Test the right deprecation message logged when custom security plugin installs rest handler
-Test no deprecation message logged when elastic Security plugin installs rest handler

elasticmachine commented:
Pinging @elastic/es-security (Team:Security)

tvernum commented on test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java:
```diff
@@ -426,6 +428,14 @@ private void ensureNoWarnings() {
         }
     }
 
+    protected void ensureNoDeprecationOnSecurityPlugin() {
```
Why do we add this to `ESTestCase` when it's only used in `SecurityPluginNoDeprecationTests` ?

BigPandaToo replied:
After refactoring `getRestHandlerWrapper` to not fail when SSLService is not created was able to move it to the SecurityTests. Reverting the Integ Test changes

tvernum commented on x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/XPackPluginTests.java:
```diff
@@ -82,8 +97,42 @@ public void testNodesNotReadyForXPackCustomMetadata() {
 
-    private XPackPlugin createXPackPlugin(Settings settings) throws Exception {
-        return new XPackPlugin(settings, null){
+    public void testCustomRestWrapperDeprecationMessage() throws Exception {
```
I don't think this test belongs here.
It should be in `ActionModuleTests`.

BigPandaToo replied:
Moved

tvernum commented on x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/SecurityPluginNoDeprecationTests.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.Collection;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class SecurityPluginNoDeprecationTests extends SecurityIntegTestCase {
```
Does this really need to be an IntegTest?
Can we not just use the same approach as the test that generates the warnings?

tvernum commented on test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java:
```diff
@@ -357,7 +357,7 @@ public static void beforeClass() throws Exception {
 
     @Override
-    protected final boolean enableWarningsCheck() {
+    protected boolean enableWarningsCheck() {
```
I don't think we should change this - it's final for a reason.

tvernum commented on x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/SecurityPluginNoDeprecationTests.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Override
+    protected boolean enableWarningsCheck() {
+        return true;
```
If we do need to keep this as an IntegTest, then you should be able to avoid needing to enable warnings by setting up a `MockLogAppender` before the node is created.

tvernum commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch '7.x' into Plugin_name_7_test

BigPandaToo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch '7.x' into Plugin_name_7_test

BigPandaToo committed:
Adding tests to #73249 change #73329

-Test the right deprecation message logged when custom security plugin installs rest handler
-Test no deprecation message logged when elastic Security plugin installs rest handler

tvernum commented on server/src/test/java/org/elasticsearch/action/ActionModuleTests.java:
```diff
@@ -196,4 +198,39 @@ public void handleRequest(RestRequest request, RestChannel channel, NodeClient c
+                TestIndexNameExpressionResolver.newInstance(),
+                moduleSettings.getIndexScopedSettings(), moduleSettings.getClusterSettings(), moduleSettings.getSettingsFilter(),
+                threadPool, singletonList(secPlugin), null, null, usageService, null);
+            assertWarnings("The org.elasticsearch.action.ActionModuleTests$7 plugin installs a custom REST wrapper. " +
```
Can we please use a named class rather than an anonymous class so that the message doesn't reference compiler-generated class names.

tvernum replied:
It doesn't look like this has changed. The plugin is still an anonymous class, so it gets a synthetic name (`$7`) instead of a real name. That means that if another test gets changed, and this plugin isn't the 7th anonymous class anymore, then the test starts to fail because the message doesn't match the synthetic name.

BigPandaToo replied:
Sorry, forgot to commit half of the change －ლ

tvernum commented on x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java:
```diff
@@ -1145,7 +1145,7 @@ public AsyncSender interceptSender(AsyncSender sender) {
         }
         final boolean ssl = HTTP_SSL_ENABLED.get(settings);
-        final SSLConfiguration httpSSLConfig = getSslService().getHttpTransportSSLConfiguration();
+        final SSLConfiguration httpSSLConfig = ssl ? getSslService().getHttpTransportSSLConfiguration() : null;
```
Please wrap all of this in an `if` rather than using the ternary operator.

tvernum reviewed (commented):
Can you update the name of the PR to something that is meaningful by itself.
The reader should not be required to know what PR #73249 is in order to know what this change is about.

BigPandaToo committed:
Merge branch '7.x' into Plugin_name_7_test

BigPandaToo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch '7.x' into Plugin_name_7_test

BigPandaToo committed:
Addressing PR feedback

tvernum commented:
> Can you update the name of the PR to something that is meaningful by itself.
> The reader should not be required to know what PR #73249 is in order to know what this change is about.

BigPandaToo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch '7.x' into Plugin_name_7_test

BigPandaToo commented:
Sorry, forgot to commit half of the change －ლ

BigPandaToo committed:
Addressing review feedback

tvernum commented on server/src/test/java/org/elasticsearch/action/ActionModuleTests.java:
```diff
@@ -234,4 +228,11 @@ public void testCustomRestWrapperDeprecationMessage() {
         }
     }
+
+    class SecPlugin implements ActionPlugin {
```
Nit: I don't think the abbreviation helps here - why `Sec` rather than `Security` ?

BigPandaToo committed:
nit cleanup

BigPandaToo commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch '7.x' into Plugin_name_7_test
